src/main/java/ca/corefacility/bioinformatics/irida/security/permissions/BasePermission.java

```diff
@@ -1,177 +1,29 @@
-	 *            the type of identifier used by this object.
-	 * @param repository
-	 *            the repository to load objects of the type for this
-	 *            permission.
-	 */
-	protected BasePermission(Class<DomainObjectType> domainObjectType, Class<IdentifierType> identifierType,
-			CrudRepository<DomainObjectType, IdentifierType> repository) {
-		this.repository = repository;
-		this.domainObjectType = domainObjectType;
-		this.identifierType = identifierType;
-	}
-
-	/**
-	 * Evaluates the permission of a single object.
-	 * 
-	 * @param authentication
-	 *            The Authentication object.
-	 * @param targetDomainObject
-	 *            The target domain object to evaluate permission (assumes this
-	 *            is not a collection).
-	 * @return True if permission is allowed on this object, false otherwise.
-	 * @throws EntityNotFoundException
-	 *             If the object does not exist.
-	 */
-	@SuppressWarnings("unchecked")
-	private boolean customPermissionAllowedSingleObject(Authentication authentication, Object targetDomainObject) {
-		DomainObjectType domainObject;
-
-		if (identifierType.isAssignableFrom(targetDomainObject.getClass())) {
-			logger.trace("Trying to find domain object by id [" + targetDomainObject + "]");
-			domainObject = repository.findById((IdentifierType) targetDomainObject).orElseThrow(() ->
-					new EntityNotFoundException("Could not find entity with id [" + targetDomainObject + "]"));
-		} else if (domainObjectType.isAssignableFrom(targetDomainObject.getClass())) {
-			// reflection replacement for instanceof
-			domainObject = (DomainObjectType) targetDomainObject;
-		} else {
-			throw new IllegalArgumentException("Parameter to " + getClass().getName() + " must be of type Long or "
-					+ domainObjectType.getName() + ".");
-		}
-		
-		/**
-		 * If it's allowed fast pass for administrators
-		 */
-		if(adminAccessAllowed(authentication, domainObject) && authentication.getAuthorities().stream().anyMatch(g -> g.getAuthority().equals(ADMIN_AUTHORITY))){
-			return true;
-		}
-
-		return customPermissionAllowed(authentication, domainObject);
-	}
-
-	/**
-	 * Tests permission for a collection of objects.
-	 * 
-	 * @param authentication
-	 *            The Authentication object.
-	 * @param targetDomainObjects
-	 *            The collection of domain objects to check for permission.
-	 * @return True if permission is allowed for every object in the collection,
-	 *         false otherwise.
-	 * @throws EntityNotFoundException
-	 *             If one of the objects in the collection does not exist.
-	 */
-	private boolean customPermissionAllowedCollection(Authentication authentication, Collection<?> targetDomainObjects) {
-		boolean permitted = true;
-		for (Object domainObjectInCollection : targetDomainObjects) {
-			permitted &= customPermissionAllowedSingleObject(authentication, domainObjectInCollection);
-		}
-
-		return permitted;
-	}
+	String getPermissionProvided();
```

Comment:
Shouldn't this have a access modifier infront of it such as public? Same thing below for `boolean isAllowed(Authentication authentication, Object targetDomainObject);`

Reply:
Since BasePermission is now an interface, all methods inside will be `public`.  We have lots of interfaces in IRIDA that specify public, but they don't need to.

Reply:
Sounds good. Thanks!